src/VirtoCommerce.Platform.Assets.AzureBlobStorage/AzureBlobProvider.cs

```diff
@@ -81,186 +86,182 @@ public virtual Stream OpenRead(string blobUrl)
-                        folder.Url = Uri.EscapeUriString(item.Uri.ToString());
-
-                        retVal.Results.Add(folder);
+                        folder.Name = item.Name.Split(Delimiter).Last();
+                        folder.Url = EscapeUri(UrlHelperExtensions.Combine(_blobServiceClient.Uri.ToString(), item.Name));
+                        result.Results.Add(folder);
                     }
-                } while (listbContinuationToken != null);
+                }
             }
 
-            retVal.TotalCount = retVal.Results.Count();
-
-            return retVal;
+            result.TotalCount = result.Results.Count();
+            return result;
         }
 
         public virtual async Task CreateFolderAsync(BlobFolder folder)
         {
-            var path = (folder.ParentUrl != null ? $"{folder.ParentUrl}/" : string.Empty) + folder.Name;
+            var path = folder.ParentUrl == null ?
+                        folder.Name :
+                        UrlHelperExtensions.Combine(folder.ParentUrl, folder.Name);
 
             var containerName = GetContainerNameFromUrl(path);
-            var blobContainer = _cloudBlobClient.GetContainerReference(containerName);
-            await blobContainer.CreateIfNotExistsAsync(BlobContainerPublicAccessType.Blob, null, null);
+            var container = _blobServiceClient.GetBlobContainerClient(containerName);
+            await container.CreateIfNotExistsAsync(PublicAccessType.Blob);
 
             var directoryPath = GetDirectoryPathFromUrl(path);
             if (!string.IsNullOrEmpty(directoryPath))
             {
                 //Need upload empty blob because azure blob storage not support direct directory creation
-                await blobContainer.GetBlockBlobReference(directoryPath).UploadFromByteArrayAsync(new byte[0], 0, 0);
+                using var stream = new MemoryStream(new byte[0]);
+                await container.GetBlockBlobClient(directoryPath).UploadAsync(stream);
             }
         }
 
         public virtual void Move(string srcUrl, string destUrl)
         {
-            MoveAsync(srcUrl, destUrl).GetAwaiter().GetResult();
+            DoMoveAsync(srcUrl, destUrl).GetAwaiter().GetResult();
+        }
+
+        public virtual Task MoveAsync(string srcUrl, string destUrl)
+        {
+            return DoMoveAsync(srcUrl, destUrl);
         }
 
         public virtual void Copy(string srcUrl, string destUrl)
         {
-            MoveAsync(srcUrl, destUrl, true).GetAwaiter().GetResult();
+            DoMoveAsync(srcUrl, destUrl, true).GetAwaiter().GetResult();
+        }
+
+        public virtual Task CopyAsync(string srcUrl, string destUrl)
+        {
+            return DoMoveAsync(srcUrl, destUrl, true);
         }
 
-        protected virtual async Task MoveAsync(string oldUrl, string newUrl, bool isCopy = false)
+        protected virtual async Task DoMoveAsync(string oldUrl, string newUrl, bool isCopy = false)
```

Comment:
Signature before changing
`protected virtual async Task MoveAsync(string oldUrl, string newUrl, bool isCopy = false)`

Signature after changing:
`public virtual Task MoveAsync(string srcUrl, string destUrl)`

Looks like breaking changes.

Reply:
MoveAsyncPublic introduced to avoid breaking changes.